### PR TITLE
[Web] Suggestion banner overflow-x support

### LINKED ISSUE #1980: [Web][iOS][Android] Displaying long suggestions
**Describe the bug**

Sometimes suggestions can be very, very long. A contrived example in English: typing "antid" may suggest "antidisestablishmentarianism":

Web/iOS interface suggesting antidisestablishmentarianism; the suggestion escapes the boundaries of the banner

For long suggestions, the suggestion breaks out of its button!

**To Reproduce**

 1. Use the attached model (change its extension to `.js`) and load it in the predictive text testing page.
 2. Type "antid" in one of the boxes.
 3. The top suggestion should overflow its button.

[simple-en-trie.txt](https://github.com/keymanapp/keyman/files/3510260/simple-en-trie.txt)

**Expected behavior**

Unsure. I think we decided back in September of last year the most appropriate behaviour is to _scroll_ long suggestions, rather than _clipping_ suggestions, or _zooming out_ long suggestions.

iOS's keyboard scrolls; SwiftKey zooms out, which we found required Antti to get out his reading glasses to be able to decipher!

Because [CSS is awesome](https://desket.co/products/css-is-awesome-mug), I think we can achieve the scrolling behaviour with the `overflow-x: scroll` declaration [[MDN]](https://developer.mozilla.org/en-US/docs/Web/CSS/overflow-x), though there may be more to it than that.

### PR DESCRIPTION
**Pull request by jahorton**
Fixes #1980.

Due to some of the requirements of the OSK, we can't rely upon native scroll handling for `overflow-x` elements, but we can still apply that styling toward the same goals if we just implement the scroll behavior ourselves.

I've tried to aim for a "natural" feel with the scroll implementation and its interactions with our normal banner-bar event handling.  **Please** live-test these changes and give feedback on the resulting UX.  I advise following the repro instructions in #1980 when testing; a great testing model is provided there.

---

Expected behaviors:

If the user's touch started within an "overflowing" suggestion:

- subsequent `touchmove` events will only scroll the suggestion
- if the user scrolls the suggestion significantly, moving a total of over 10 units, releasing the touch will _not_ select the suggestion.
  - However, if the total is under 10 units, the suggestion _will_ be selected.
  - While the exact number (10 units) could be tweaked, the core idea is that a bit of robustness against noise will be needed to allow overflowing suggestions to "feel right" and be selectable when intended.

In testing, it personally "felt wrong" to have the suggestion accepted after a long scroll.  So, I've implemented scrolling and selection as two separate states / UX "commands."

If the user's touch started outside of an overflowing suggestion:

- As before, `touchmove`s to the left or right will highlight the closest suggestion.
- if the newly-closest suggestion is overflowing, it *will not be scrolled*.
  - Basically,  the banner will automatically be in "selection" mode, not "scroll" mode, when started in this manner.

---

Correspondingly, we could say that "scroll" mode is formally activated when a user starts a touch within an overflow element and `touchmove`s at least 10 units total; before that, we interpret the command as a "selection" instead.

### CODE REVIEW

**Comment by darcywong00** [comment]
Posting a screenshot on Android. I didn't use the custom model in 1980, but you can scroll through the 2 suggestions "antidespressanst" and "antidespressant"
Screenshot_1568090399

File: web/source/dom/uiTouchHandlerBase.ts [hunk 1]
```diff
@@ -249,6 +285,9 @@ namespace com.keyman.dom {
       var x = e.changedTouches[0].pageX;
       var beyondEdge = ((x < 2 && this.touchX > 5) || (x > window.innerWidth - 2 && this.touchX < window.innerWidth - 5));
 
+      // Allow an accidental fudge-factor for overflow element noise, but not much.
+      beyondEdge = beyondEdge || this.scrollTouchState.totalLength > 10;
```
Comment (mcdurdin) [inline web/source/dom/uiTouchHandlerBase.ts]: Please make the `10` into a constant; I know we've got older code that doesn't religiously use constants, but I'd like to try and avoid perpetuating this :)

Comment (eddieantonio) [reply web/source/dom/uiTouchHandlerBase.ts]: Seconding this! Please make it a constant like... `EDGE_FUDGE_FACTOR`...?

File: web/source/dom/uiTouchHandlerBase.ts [hunk 2]
```diff
@@ -205,6 +237,10 @@ namespace com.keyman.dom {
         return;
       }
 
+      // Establish scroll tracking.
+      let scrollMode = (this.currentTarget.clientWidth < this.currentTarget.scrollWidth);
```
Comment (eddieantonio) [inline web/source/dom/uiTouchHandlerBase.ts]: I suggest changing this name to `shouldScroll`; then the line after this reads as "if shouldScroll then make a touch state"
